src/python/pants/backend/python/target_types_rules.py

```diff
@@ -143,29 +153,160 @@ async def inject_pex_binary_entry_point_dependency(
 # -----------------------------------------------------------------------------------------------
 
 
+def _classify_entry_points(
+    entry_points: FrozenDict[str, FrozenDict[str, str]]
+) -> Generator[Tuple[bool, str, str, str], None, None]:
+    """Looks at each entry point to see if it is a target address or not.
+
+    Returns a tuple of tuples: IsTarget, section, name, ref.
+    """
+    for section, values in entry_points.items():
+        for name, ref in values.items():
+            yield ref.startswith(":") or "/" in ref, section, name, ref
+
+
+@rule(desc="Determining the entry points for a `python_distribution` target")
```

Comment:
This isn't documented at all...you'll have to add `level=LogLevel.DEBUG` to this for it to show up in the dynamic UI. Otherwise the description gets kind of discarded.

Imo, this is a great description you wrote and helpful to be in the dynamic UI :)

Reply:
copy paste from `resolve_pex_entry_point`. I'll add the level there too, then..